compiler/rustc_const_eval/src/interpret/eval_context.rs

```diff
@@ -694,9 +694,10 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
                 let elem = layout.field(self, 0);
 
                 // Make sure the slice is not too big.
-                let size = elem.size.checked_mul(len, self).ok_or_else(|| {
-                    err_ub!(InvalidMeta("slice is bigger than largest supported object"))
-                })?;
+                let size = elem.size * len;
```

Comment:
Hm, actually... this multiplication could panic, which wouldn't be good. But `Size::checked_mul` checks the wrong bound for us. Not quite sure what the best thing to do is here. (This also affects other places where we use `Size::checked_mul`.)

Reply:
> (This also affects other places where we use Size::checked_mul.)

Actually, the other places all do real memory accesses of that size, so the check is okay for them given that we can never have an actual allocation of that size.

Reply:
The latest commit fixes the overflow.